Extraneous ghost lines on PathLayer renders

It seems that when trying to render a path using the fp64 flag, there are ghostlines that extrude from the turns on these paths. The behavior seems to go away once you turn off fp64 though.

Here's an example:
![screenshot 2017-08-30 12 33 15](https://user-images.githubusercontent.com/836037/29892846-fcdad9b2-8d84-11e7-9d9f-9bbfbeb38518.png)

Reproduction Steps:
```
new PathLayer({
    fp64: true,
    ...pathLayerProps
});
```

Version tested on: 4.1.1

Graphics  @Processor Specs:
```
$ system_profiler SPDisplaysDataType
Graphics/Displays:

    Intel Iris Pro:

      Chipset Model: Intel Iris Pro
      Type: GPU
      Bus: Built-In
      VRAM (Dynamic, Max): 1536 MB
      Vendor: Intel (0x8086)
      Device ID: 0x0d26
      Revision ID: 0x0008
      Automatic Graphics Switching: Supported
      gMux Version: 4.0.8 [3.2.8]
      Metal: Supported

    NVIDIA GeForce GT 750M:

      Chipset Model: NVIDIA GeForce GT 750M
      Type: GPU
      Bus: PCIe
      PCIe Lane Width: x8
      VRAM (Total): 2048 MB
      Vendor: NVIDIA (0x10de)
      Device ID: 0x0fe9
      Revision ID: 0x00a2
      ROM Revision: 3776
      Automatic Graphics Switching: Supported
      gMux Version: 4.0.8 [3.2.8]
      Metal: Supported
```

cc: @ankitmehta21 @maxijb

@shaojingli , how did you reproduce this, were you using `gps-trace-changes` branch?

I am getting errors with `npm install` or `yarn` :


```
chandu-C02RG1F9G8WM:trip-debugger raviakkenapally$ npm install
npm ERR! Darwin 16.0.0
npm ERR! argv "/Users/raviakkenapally/.nvm/versions/node/v7.10.0/bin/node" "/Users/raviakkenapally/.nvm/versions/node/v7.10.0/bin/npm" "install"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! code E404

npm ERR! 404 Not found : @uber/build-tasks
npm ERR! 404 
npm ERR! 404  '@uber/build-tasks' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'trip-debugger'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/raviakkenapally/.npm/_logs/2017-08-31T17_07_53_219Z-debug.log
```

yarn:

```
yarn install v0.24.5
info No lockfile found.
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.npmjs.org/@uber%2feslint-config-jsx: Not found".
info If you think this is a bug, please open a bug report with the information provided in "/Users/raviakkenapally/Documents/Source/DataViz/trip-debugger/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

```

--- a/src/core-layers/path-layer/path-layer-vertex-64.glsl.js
+++ b/src/core-layers/path-layer/path-layer-vertex-64.glsl.js
@@ -50,6 +50,7 @@ varying float vPathPosition;
 varying float vPathLength;
 
 const float EPSILON = 0.001;
+const float PIXEL_EPSILON = 0.1;
 
 float flipIfTrue(bool flag) {
   return -(float(flag) * 2. - 1.);
@@ -79,8 +80,12 @@ vec3 lineJoin(vec2 prevPoint64[2], vec2 currPoint64[2], vec2 nextPoint64[2]) {
   float offsetScale;
   float offsetDirection;
 
-  vec2 dirA = lenA > 0. ? deltaA / lenA : vec2(1.0, 0.0);
-  vec2 dirB = lenB > 0. ? deltaB / lenB : vec2(1.0, 0.0);
+  // when two points are closer than PIXEL_EPSILON in pixels,
+  // assume they are the same point to avoid precision issue
+  lenA = lenA > PIXEL_EPSILON ? lenA : 0.0;
+  lenB = lenB > PIXEL_EPSILON ? lenB : 0.0;
+  vec2 dirA = lenA > 0. ? deltaA / lenA : vec2(0.0, 0.0);
+  vec2 dirB = lenB > 0. ? deltaB / lenB : vec2(0.0, 0.0);
 
   vec2 perpA = vec2(-dirA.y, dirA.x);
   vec2 perpB = vec2(-dirB.y, dirB.x);

